### ISSUE TITLE
menu missing

### ISSUE THREAD

**Issue by SzVLsY** [pionxzh/chatgpt-exporter#251]
<img width="263" alt="image" src="https://github.com/user-attachments/assets/2c68a445-1658-4f0c-8be1-0ca136c469c1">

**Comment by SzVLsY** [comment]
Thank you. I have found the issue. Enabling Dark Reader causes the menu to disappear. Simply adding the domain to the whitelist and refreshing the page will restore everything to normal.  
https://github.com/darkreader/darkreader

This is the value of `accounts.default.features` when Dark Reader is enabled.
```


[
  "arkose_enabled",
  "arkose_enabled_for_subscription",
  "beta_features",
  "bizmo_settings",
  "breeze_available",
  "browsing_available",
  "chart_serialization",
  "chat_preferences_available",
  "chatgpt_ios_attest",
  "chatgpt_team_plan",
  "code_interpreter_available",
  "d3_controls",
  "d3_editor",
  "d3_editor_gpts",
  "dalle_3",
  "gizmo_reviews",
  "gizmo_support_emails",
  "graphite",
  "invite_referral",
  "memory_ga",
  "mfa",
  "model_ab_use_v2",
  "model_switcher",
  "new_plugin_oauth_endpoint",
  "paragen_mainline_alternative",
  "plugins_available",
  "privacy_policy_nov_2023",
  "sentinel_enabled_for_subscription",
  "shareable_links",
  "starter_prompts",
  "sunshine_available",
  "user_settings_announcements"
]
```

**Comment by SzVLsY** [comment]
> Can you help to open network panel in devtool, find `backend-api/accounts/check/v4-2023-04-27` API request
> 
> And then copy the value of `accounts.default.features`
> 
> ![photo_2024-09-02_19-25-47](https://private-user-images.githubusercontent.com/9910706/363672182-4a129cbf-18c0-46bf-94f6-b8352e8b987b.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUyNzgzMjgsIm5iZiI6MTcyNTI3ODAyOCwicGF0aCI6Ii85OTEwNzA2LzM2MzY3MjE4Mi00YTEyOWNiZi0xOGMwLTQ2YmYtOTRmNi1iODM1MmU4Yjk4N2IuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDJUMTE1MzQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzI4ZTcxNTNmOWMzYmU3YTk4Nzc3MWU0YzhjNzBiY2VjNjM5YmYwYjZlZDJkMTFlNzg0NGRiNDBjMmI4YmQ1OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.kXFCWQANVM8X_R9xfHHLCkAksX6P0x5YXPX6oPwhLa0)

Thank you. I have found the issue. Enabling Dark Reader causes the menu to disappear. Simply adding the domain to the whitelist and refreshing the page will restore everything to normal.  
https://github.com/darkreader/darkreader

This is the value of `accounts.default.features` when Dark Reader is enabled.
```


[
  "arkose_enabled",
  "arkose_enabled_for_subscription",
  "beta_features",
  "bizmo_settings",
  "breeze_available",
  "browsing_available",
  "chart_serialization",
  "chat_preferences_available",
  "chatgpt_ios_attest",
  "chatgpt_team_plan",
  "code_interpreter_available",
  "d3_controls",
  "d3_editor",
  "d3_editor_gpts",
  "dalle_3",
  "gizmo_reviews",
  "gizmo_support_emails",
  "graphite",
  "invite_referral",
  "memory_ga",
  "mfa",
  "model_ab_use_v2",
  "model_switcher",
  "new_plugin_oauth_endpoint",
  "paragen_mainline_alternative",
  "plugins_available",
  "privacy_policy_nov_2023",
  "sentinel_enabled_for_subscription",
  "shareable_links",
  "starter_prompts",
  "sunshine_available",
  "user_settings_announcements"
]
```

**Comment by SzVLsY** [comment]
In Firefox, aside from Dark Reader, Privacy Badger also seems to cause the menu to disappear.

**Comment by michaelskyba** [comment]
My menu is also missing. I'm using a Chromium-based browser and without Dark Reader. My features list is exactly the same as SzVLsY's, though.

**Comment by michaelskyba** [comment]
I'm using Arch Linux 6.9.9-arch1-1 with Brave "1.68.141 Chromium: 127.0.6533.120 (Official Build) (64-bit)". I also have a different browser installed: base Chromium "126.0.6478.182 (Official Build) Arch Linux (64-bit)" installed.

For both of my two OpenAI accounts, the Export button does not show on Brave, and for both accounts, the Export button still shows on Chromium. In both cases, I'm only using Tampermonkey. Both Chromium and Brave were using the same v4-2023-04-27 file.

I could try doing some further debugging on my end, including trying Firefox, if I have time later

**Comment by pionxzh** [comment]
I don't have Arch Linux environment, but I will be able to test it on Mac M-chip tomorrow.

**Comment by michaelskyba** [comment]
In at least one of my Brave tests I had no extensions enabled besides Tampermonkey, so I don't think any others are interfering, but I could try private mode too later. Besides macOS, maybe also try testing Brave on Windows 11, in case that's the problem

**Comment by pionxzh** [comment]
Hey, can you check this?
I think they are A/B testing the `remix` version.

![image](https://github.com/user-attachments/assets/17b340a9-b928-4f93-830d-5b73f7d03ad8)

source: https://twitter.com/btibor91/status/1828342132643332514

**Comment by michaelskyba** [comment]
Both Brave and Chromium have ``<html data-build="remix-prod-(hash...)`` in the source for me. Both have the same hash, starting with b13f36.

However, I apologize, as I've realized that some of my previous comments were misleading. I did previously attempt to run the exporter script without any other extensions active, but I see now that Tampermonkey wasn't running correctly there, because that profile had Developer Mode off.

In my other profile, with more extensions, I did some tests and it looks like the Export button disappears when the [Stylus](https://add0n.com/stylus.html) extension is active, even if the injected stylesheet I configured doesn't apply to any elements on the page.

It still might be related to Remix because even with Stylus active, ChatGPT Exporter has been working up until a few days ago

**Comment by pionxzh** [comment]
@rndwalker maybe you can check these
1. https://www.torn.com/forums.php#/p=threads&f=67&t=16255988&b=0&a=0
2. https://github.com/lisonge/vite-plugin-monkey/issues/1

CSP issue cannot be fixed on my end. 🙏

**Comment by pionxzh** [comment]
@rndwalker But im curious, when did you start to notice that the export button disappeared?

**Comment by pionxzh** [comment]
@rndwalker Not sure what's your enviroument, but I can successfully see my script on a brand new chrome profile with only Tampermonkey and this script. I do see other scripts triggered the CSP error on my PC after disabling the force remove CSP restricition options, but this `chatgpt-exporter` doesn't require that option to be enabled. So maybe you can try disable other scripts and test again?

**Comment by pionxzh** [comment]
I assume everyone here seems hit on different walls to see the menu.
1. CSP issues
2. Tampermonkey not enabled / haven't follow the steps after updating to MF3
3. The script itself do have issue in some scienrio

During the testing, I noticed that the menu will disappear if the screen is not big enough to see the left panel on page load. I will release a fix and try to enhance the menu injection to avoid any other scripts/extentions to corrupt the menu.

**Comment by pionxzh** [comment]
I just released v2.24.5. Please help to see if it fixes the issue.

**Comment by pionxzh** [comment]
Based on https://github.com/pionxzh/chatgpt-exporter/pull/252#issuecomment-2336761125, I assume this issue has been fixed. I will close the issue. Let me know if you still cannot see the menu.

**Comment by SzVLsY** [comment]
> I just released v2.24.5. Please help to see if it fixes the issue.

I tested the new version, and when I enable Dark Reader, the menu disappears again. Additionally, the menu also disappears when I enable some scripts. However I would say there is some improvement, as I no longer need to completely disable the extensions; I just need to add them to the whitelist. Thank you for your work.

**Comment by SzVLsY** [comment]
> Can anyone share a combination that can reproduce the issue? I tried these, all work correctly.
> 
>     * Windows 11, Chrome 127
> 
>     * Windows 11, Edge 128
> 
>     * Windows 11. Firefox 130 (With `Privacy Badger`)
>       ![image](https://private-user-images.githubusercontent.com/9910706/364039097-8ee8921d-a370-4e1d-8e08-0816c1ca5e16.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4ODkxOTEsIm5iZiI6MTcyNTg4ODg5MSwicGF0aCI6Ii85OTEwNzA2LzM2NDAzOTA5Ny04ZWU4OTIxZC1hMzcwLTRlMWQtOGUwOC0wODE2YzFjYTVlMTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDlUMTMzNDUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWJhNDI1YTg5MjdjODlhNjI1OTkxMDU2YTQwOGRjZGEwOWY3NDg2YWZjYzFkOWExZWIzZTIyYTRhOGIyMmY0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.op_bm-Rqw3LhwazWoWFT9IkGxEhwZkkbsRDZQIHZC-U)
> 
> 
> All tested with the dark reader on/off
> 
> btw, if you are using userscript manager other than Tampermonkey, please also let me know

On Windows 10 with Chrome 129 and version 2.24.6, the menu disappears after enabling Dark Reader.

Sometimes, the menu works fine one day and suddenly disappears the next. When I tried installing the PR version I compiled, everything was fine the first night, but the menu disappeared again after a day. The browser extensions and scripts were not updated.

**Comment by pionxzh** [comment]
> when I enable Dark Reader, the menu disappears again
That's weird. I also tested it with Dark Reader, but it works totally fine.

> I just need to add them to the whitelist
Can you elaborate more on what whitelist you changed?

**Comment by SzVLsY** [comment]
> > when I enable Dark Reader, the menu disappears again
> 
> That's weird. I also tested it with Dark Reader, but it works totally fine.
> 
> > I just need to add them to the whitelist
> 
> Can you elaborate more on what whitelist you changed?

<img width="267" alt="image" src="https://github.com/user-attachments/assets/f61ea941-bcd1-4ea5-980c-b8f5b5346012">

**Comment by SzVLsY** [comment]
> > when I enable Dark Reader, the menu disappears again
> 
> That's weird. I also tested it with Dark Reader, but it works totally fine.
> 
> > I just need to add them to the whitelist
> 
> Can you elaborate more on what whitelist you changed?

Update: 
Today, both versions completely stopped working. This is really strange—I simply refreshed the page, and then the menu disappeared. I've tried restarting the browser, and today I disabled all extensions and scripts except for the export script, but that didn't work.

**Comment by SzVLsY** [comment]
> > when I enable Dark Reader, the menu disappears again
> 
> That's weird. I also tested it with Dark Reader, but it works totally fine.
> 
> > I just need to add them to the whitelist
> 
> Can you elaborate more on what whitelist you changed?

Today, I noticed that when I refresh the page, the menu appears, and then when the page finishes loading, the menu disappears. Version 2.24.6.  
When I reinstall version 2.24.4, the menu appears fixed in the top left corner, but when I refresh the page, the menu disappears. I have to resize the window again.

<img width="1043" alt="image" src="https://github.com/user-attachments/assets/323b5aad-f4f0-4812-9d0b-528434e59a9f">

**Comment by pionxzh** [comment]
Thanks for the verification. I'm trying to fix it in #258

**Comment by 14790897** [comment]
The Remix framework must be doing this. It constantly monitors the elements it manages, and if any changes are detected, it immediately reverts them back.
